### PR TITLE
r/efs_file_system - add attributes + refactor to use waiter package

### PR DESCRIPTION
**Pull request by DrFaust92**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccAWSEFSFileSystem_'
--- PASS: TestAccAWSEFSFileSystem_kmsConfigurationWithoutEncryption (28.50s)
--- PASS: TestAccAWSEFSFileSystem_pagedTags (53.94s)
--- PASS: TestAccAWSEFSFileSystem_disappears (53.99s)
--- PASS: TestAccAWSEFSFileSystem_kmsKey (78.62s)
--- PASS: TestAccAWSEFSFileSystem_ProvisionedThroughputInMibps (90.30s)
--- PASS: TestAccAWSEFSFileSystem_lifecyclePolicy (94.44s)
--- PASS: TestAccAWSEFSFileSystem_ThroughputMode (94.75s)
--- PASS: TestAccAWSEFSFileSystem_basic (103.86s)
--- PASS: TestAccAWSEFSFileSystem_lifecyclePolicy_removal (107.29s)
--- PASS: TestAccAWSEFSFileSystem_tags (180.87s)
--- PASS: TestAccAWSEFSFileSystem_lifecyclePolicy_update (183.95s)
```

### CODE REVIEW

File: aws/resource_aws_efs_file_system.go [hunk 1]
```diff
@@ -208,21 +213,11 @@ func resourceAwsEfsFileSystemUpdate(d *schema.ResourceData, meta interface{}) er
 
 		_, err := conn.UpdateFileSystem(input)
 		if err != nil {
-			return fmt.Errorf("error updating EFS File System %q: %s", d.Id(), err)
+			return fmt.Errorf("error updating EFS File System %q: %w", d.Id(), err)
```
Comment (ewbankkit) [inline aws/resource_aws_efs_file_system.go:216]: For consistency.
```suggestion
			return fmt.Errorf("error updating EFS file system (%s): %w", d.Id(), err)
```

File: aws/resource_aws_efs_file_system.go [hunk 2]
```diff
@@ -335,77 +328,29 @@ func resourceAwsEfsFileSystemDelete(d *schema.ResourceData, meta interface{}) er
...
 	}
 
-	err = waitForDeleteEfsFileSystem(conn, d.Id(), 10*time.Minute)
-	if err != nil {
-		return fmt.Errorf("Error waiting for EFS file system (%q) to delete: %w", d.Id(), err)
+	if _, err := waiter.FileSystemDeleted(conn, d.Id()); err != nil {
+		if isAWSErr(err, efs.ErrCodeFileSystemNotFound, "") {
+			return nil
+		}
+		return fmt.Errorf("error waiting for EFS access point (%s) deletion: %w", d.Id(), err)
```
Comment (ewbankkit) [inline aws/resource_aws_efs_file_system.go:341]:
```suggestion
		return fmt.Errorf("error waiting for EFS file system (%s) deletion: %w", d.Id(), err)
```

File: aws/resource_aws_efs_file_system.go [hunk 3]
```diff
@@ -316,12 +309,12 @@ func resourceAwsEfsFileSystemRead(d *schema.ResourceData, meta interface{}) erro
 		FileSystemId: fs.FileSystemId,
 	})
 	if err != nil {
-		return fmt.Errorf("Error describing lifecycle configuration for EFS file system (%s): %s",
+		return fmt.Errorf("Error describing lifecycle configuration for EFS file system (%s): %w",
 			aws.StringValue(fs.FileSystemId), err)
```
Comment (ewbankkit) [inline aws/resource_aws_efs_file_system.go:313]:
```suggestion
		return fmt.Errorf("Error describing lifecycle configuration for EFS file system (%s): %w", d.Id(), err)
```

File: aws/resource_aws_efs_file_system.go [hunk 4]
```diff
@@ -316,12 +309,12 @@ func resourceAwsEfsFileSystemRead(d *schema.ResourceData, meta interface{}) erro
 		FileSystemId: fs.FileSystemId,
```
Comment (ewbankkit) [inline aws/resource_aws_efs_file_system.go:309]: For consistency.
```suggestion
		FileSystemId: aws.String(d.Id()),
```

**Comment by ewbankkit** [comment]
Verified acceptance tests:

```console
$ make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSEFSFileSystem_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSEFSFileSystem_ -timeout 120m
=== RUN   TestAccAWSEFSFileSystem_basic
=== PAUSE TestAccAWSEFSFileSystem_basic
=== RUN   TestAccAWSEFSFileSystem_tags
=== PAUSE TestAccAWSEFSFileSystem_tags
=== RUN   TestAccAWSEFSFileSystem_pagedTags
=== PAUSE TestAccAWSEFSFileSystem_pagedTags
=== RUN   TestAccAWSEFSFileSystem_kmsKey
=== PAUSE TestAccAWSEFSFileSystem_kmsKey
=== RUN   TestAccAWSEFSFileSystem_kmsConfigurationWithoutEncryption
=== PAUSE TestAccAWSEFSFileSystem_kmsConfigurationWithoutEncryption
=== RUN   TestAccAWSEFSFileSystem_ProvisionedThroughputInMibps
=== PAUSE TestAccAWSEFSFileSystem_ProvisionedThroughputInMibps
=== RUN   TestAccAWSEFSFileSystem_ThroughputMode
=== PAUSE TestAccAWSEFSFileSystem_ThroughputMode
=== RUN   TestAccAWSEFSFileSystem_lifecyclePolicy
=== PAUSE TestAccAWSEFSFileSystem_lifecyclePolicy
=== RUN   TestAccAWSEFSFileSystem_lifecyclePolicy_update
=== PAUSE TestAccAWSEFSFileSystem_lifecyclePolicy_update
=== RUN   TestAccAWSEFSFileSystem_lifecyclePolicy_removal
=== PAUSE TestAccAWSEFSFileSystem_lifecyclePolicy_removal
=== RUN   TestAccAWSEFSFileSystem_disappears
=== PAUSE TestAccAWSEFSFileSystem_disappears
=== CONT  TestAccAWSEFSFileSystem_basic
=== CONT  TestAccAWSEFSFileSystem_ThroughputMode
=== CONT  TestAccAWSEFSFileSystem_disappears
=== CONT  TestAccAWSEFSFileSystem_lifecyclePolicy_removal
=== CONT  TestAccAWSEFSFileSystem_lifecyclePolicy_update
=== CONT  TestAccAWSEFSFileSystem_lifecyclePolicy
=== CONT  TestAccAWSEFSFileSystem_kmsKey
=== CONT  TestAccAWSEFSFileSystem_ProvisionedThroughputInMibps
=== CONT  TestAccAWSEFSFileSystem_kmsConfigurationWithoutEncryption
=== CONT  TestAccAWSEFSFileSystem_pagedTags
=== CONT  TestAccAWSEFSFileSystem_tags
--- PASS: TestAccAWSEFSFileSystem_kmsConfigurationWithoutEncryption (17.69s)
--- PASS: TestAccAWSEFSFileSystem_kmsKey (61.96s)
--- PASS: TestAccAWSEFSFileSystem_lifecyclePolicy_update (70.51s)
--- PASS: TestAccAWSEFSFileSystem_basic (74.90s)
--- PASS: TestAccAWSEFSFileSystem_pagedTags (99.69s)
--- PASS: TestAccAWSEFSFileSystem_ThroughputMode (101.77s)
--- PASS: TestAccAWSEFSFileSystem_lifecyclePolicy (104.60s)
--- PASS: TestAccAWSEFSFileSystem_lifecyclePolicy_removal (106.78s)
--- PASS: TestAccAWSEFSFileSystem_ProvisionedThroughputInMibps (128.89s)
--- PASS: TestAccAWSEFSFileSystem_disappears (134.35s)
--- PASS: TestAccAWSEFSFileSystem_tags (140.15s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	140.282s
```

**Review by bflad — APPROVED** [review]
Looks good to me, thank you, @DrFaust92 🚀 

Output from acceptance testing in AWS Commercial:

```
--- PASS: TestAccAWSEFSFileSystem_kmsConfigurationWithoutEncryption (15.26s)
--- PASS: TestAccAWSEFSFileSystem_lifecyclePolicy (29.48s)
--- PASS: TestAccAWSEFSFileSystem_disappears (37.95s)
--- PASS: TestAccAWSEFSFileSystem_kmsKey (54.93s)
--- PASS: TestAccAWSEFSFileSystem_ThroughputMode (65.88s)
--- PASS: TestAccAWSEFSFileSystem_basic (83.23s)
--- PASS: TestAccAWSEFSFileSystem_pagedTags (85.76s)
--- PASS: TestAccAWSEFSFileSystem_ProvisionedThroughputInMibps (94.41s)
--- PASS: TestAccAWSEFSFileSystem_lifecyclePolicy_removal (96.71s)
--- PASS: TestAccAWSEFSFileSystem_lifecyclePolicy_update (107.04s)
--- PASS: TestAccAWSEFSFileSystem_tags (142.76s)
```

Output from acceptance testing in AWS GovCloud (US):

```
--- PASS: TestAccAWSEFSFileSystem_kmsConfigurationWithoutEncryption (19.51s)
--- PASS: TestAccAWSEFSFileSystem_pagedTags (58.12s)
--- PASS: TestAccAWSEFSFileSystem_kmsKey (64.15s)
--- PASS: TestAccAWSEFSFileSystem_lifecyclePolicy (67.15s)
--- PASS: TestAccAWSEFSFileSystem_disappears (85.41s)
--- PASS: TestAccAWSEFSFileSystem_basic (95.51s)
--- PASS: TestAccAWSEFSFileSystem_ThroughputMode (101.41s)
--- PASS: TestAccAWSEFSFileSystem_ProvisionedThroughputInMibps (103.19s)
--- PASS: TestAccAWSEFSFileSystem_lifecyclePolicy_update (105.87s)
--- PASS: TestAccAWSEFSFileSystem_tags (149.67s)
--- PASS: TestAccAWSEFSFileSystem_lifecyclePolicy_removal (151.51s)
```

**Pull request merged**